pysrc/bytewax/connectors/kafka/serde.py

```diff
@@ -0,0 +1,94 @@
+"""Serializers and deserializers for kafka messages."""
+import io
+import json
+import logging
+from abc import ABC, abstractmethod
+
+from confluent_kafka.schema_registry.avro import AvroDeserializer, AvroSerializer
+from confluent_kafka.serialization import (
+    MessageField,
+    SerializationContext,
+)
+from fastavro import parse_schema, schemaless_reader, schemaless_writer
+
+logger = logging.getLogger(__name__)
+
+
+class SerializationError(ValueError):
+    """(de)serialization exception."""
+
+
+class SchemaSerializer(ABC):
+    """A serializer for a specific schema."""
+
+    @abstractmethod
+    def ser(self, obj: dict, topic: str) -> bytes:
+        """Serialize data.
+
+        The `topic` argument can be ignored if not needed.
+
+        Do not catch serialization errors, let the source handle it.
+        """
+        ...
+
+
+class SchemaDeserializer(ABC):
+    """A deserializer for a specific schema."""
+
+    @abstractmethod
+    def de(self, data: bytes, topic: str) -> dict:
+        """Deserialize data.
+
+        The `topic` argument can be ignored if not needed.
+
+        Do not catch deserialization errors, let the source handle it.
+
+        """
+        ...
+
+
+class _ConfluentAvroSerializer(SchemaSerializer):
+    def __init__(self, client, schema_str, is_key):
+        self.serializer = AvroSerializer(client, schema_str)
+        self.ctx = SerializationContext(
+            # This value will be updated at each message
+            "PLACEHOLDER",
```

Comment:
Confluent's api require both the field and the topic info each time you call the (de)serialization function, so at first I was initializing the ctx object at each message. I moved the initialization here but put a placeholder topic since the topic was assigned at every message, but yep, since our topic shouldn't change after initialization, I'm moving that up so we don't need to do this